srtcore/core.cpp

```diff
@@ -8191,19 +8191,34 @@ void CUDT::processCtrl(const CPacket &ctrlpkt)
     case UMSG_ACKACK: // 110 - Acknowledgement of Acknowledgement
     {
         int32_t ack = 0;
-        const int rtt = m_ACKWindow.acknowledge(ctrlpkt.getAckSeqNo(), ack);
+
+        // Calculate RTT estimate on the receiver side based on ACK/ACKACK pair
+        const int rtt = m_ACKWindow.acknowledge(ctrlpkt.getAckSeqNo(), ack, currtime);
+
+        if (rtt == -1)
+        {
+            LOGC(inlog.Error,
+                 log << CONID() << "IPE: The record about ACK is not found, "
+                     << "RTT estimate at the receiver side can not be calculated "
+                     << "(ACK number: " << ctrlpkt.getAckSeqNo() << ", last ACK sent: " << m_iAckSeqNo
+                     << ", oldest ACK record: " << "not yet available" << ", RTT (EWMA): " << m_iRTT << ")");
```

Comment:
Consider shortening the log message.

```suggestion
            LOGC(inlog.Error,
                 log << CONID() << "IPE: ACK record not found, "
                     << "RTT estimate impossible "
                     << "(ACK number: " << ctrlpkt.getAckSeqNo() << ", last ACK sent: " << m_iAckSeqNo
                     << ", RTT (EWMA): " << m_iRTT << ")");
```